packages/components/app/styles/components/application-state.scss

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+//
+// APPLICATION-STATE COMPONENT
+//
+$hds-application-state-padding: 12px 0;
+
+.hds-application-state {
+  align-self: center;
+  width: 20rem;
+  max-width: 100%;
+  margin: 0 auto;
+}
+
+.hds-application-state__header,
+.hds-application-state__body,
+.hds-application-state__footer {
+  display: flex;
+}
+
+.hds-application-state__header {
+  display: grid;
+  grid-row-gap: 0;
+  grid-column-gap: 0;
+  color: var(--token-color-foreground-primary);
+
+  .hds-application-state--error & {
+    grid-template-rows: repeat(2, auto);
+    grid-template-columns: 28px auto;
+    align-items: stretch;
+
+    .hds-application-state__header-icon {
+      grid-area: 1 / 1 / 3 / 2;
+      padding-top: 4px;
+    }
+
+    .hds-application-state__header-title {
+      grid-area: 1 / 2 / 2 / 3;
+    }
+
+    .hds-application-state__header-errorCode {
+      grid-area: 2 / 2 / 3 / 3;
+    }
+  }
+}
+
+.hds-application-state__body {
+  padding: $hds-application-state-padding;
+  color: var(--token-color-foreground-faint);
+}
+
+.hds-application-state__footer {
+  display: flex;
+  justify-content: space-between;
+
+  .hds-application-state--has-divider & {
+    padding: $hds-application-state-padding;
+    border-top: 1px solid gray;
```

Comment:
Should check if we have a standardized color token to match the desired border color.

Reply:
The border should be using `--token-color-border-strong`

cc @andgen404